components/automate-ui/src/app/pages/user-details/user-details.component.ts

```diff
@@ -45,63 +44,62 @@ export class UserDetailsComponent implements OnInit, OnDestroy {
     private fb: FormBuilder
   ) {
     this.createForms(this.fb);
-    // TODO (tc) This needs to be refactored to resemble our other patterns
-    // for specific object pages.
-    // combineLatest depends on the user object existing already.
-    this.user = {
-      id: '',
-      name: '',
-      membership_id: ''
-    };
-
-    this.done$ = <Observable<boolean>>combineLatest([
-      store.select(allUsers),
-      store.select(userStatus)])
-      .pipe(
-        map(([state, status]: [User[], EntityStatus]) => {
-          const id = this.user.id;
-          return status === EntityStatus.loadingSuccess && !find({ id }, state);
-        }));
-
-    this.subscriptions = [
-      store.select(userFromRoute).pipe(
-        filter(identity))
-        .subscribe((state) => {
-          this.user = <User>state;
-        }),
-      // Note: if the user browses directly to /settings/users/USERNAME, the state
-      // will not contain any user data -- so we need to fetch it.
-      store.select(routeParams).pipe(
-        pluck('id'),
-        filter(identity))
-        .subscribe((id: string) => {
-          store.dispatch(new GetUser({id}));
-        }),
-
-      // if the user is gone, go back to list
-      // note: we filter(identity) here -- so this will effectively only
-      // trigger when done$ is `true`
-      this.done$.pipe(filter(identity)).subscribe(
-        () => this.router.navigate(['/settings', 'users'])),
-
-      store.select(updateStatus).subscribe(this.handleUpdateStatus.bind(this))
-      ];
     }
 
   ngOnInit(): void {
-    this.route.data.subscribe((data: { isNonAdmin: boolean }) => {
-      this.isAdminView = !data.isNonAdmin;
-      // Update the forms once this info has come in
-      this.createForms(this.fb);
-      this.resetForms();
+    this.store.pipe(
+      select(routeParams),
+      pluck('id'),
+      filter(identity),
+      takeUntil(this.isDestroyed))
+      .subscribe((id: string) => {
+        this.store.dispatch(new GetUser({ id }));
+      });
+
+    combineLatest([
+      this.store.select(getStatus),
+      this.store.select(userFromRoute)
+    ]).pipe(
+      filter(([status, user]) => status === EntityStatus.loadingSuccess && !isNil(user)),
+      takeUntil(this.isDestroyed))
+      .subscribe(([_, user]) => {
+        this.user = { ...user };
```

Comment:
What does the `{ ...user }` do?

Reply:
that's the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), which is a shorthand for cloning objects

Reply:
Oh, I did not know that cloned an object. That is a nice shorthand to know.